setup.py

```diff
@@ -1,11 +1,17 @@
-from setuptools import setup, find_packages
+import pathlib
+
+from setuptools import find_packages, setup
+
+HERE = pathlib.Path(__file__).parent
+README = (HERE / "README.md").read_text()
 
 setup(
     name='PyISY',
-    version='1.1.1',
+    version='1.2.0',
```

Comment:
This should probably be a version 2.y.z. There aren't really changes to the external API, but it is such a big refactoring that it would probably be good to track the change.

Reply:
Agreed. I had planned on bumping the major version before publishing this update with a release page writeup of why.

Reply:
👍

Reply:
👍

Reply:
A bigger question to this--if this is going to be a major version bump, should I go ahead and finish updating the function names/properties to be PEP8 compliant? I left several "externally called" functions the same to not be a breaking change (e.g. `sendX10`), but if we document the changes it wouldn't be much more to fix the rest of the functions.  My original intent was to not introduce any major breaking changes--but there are a few.

Reply:
Good question. Let's keep the method names the same with the goal of this update having no "known" big breaking changes. This will increase the number of people that will update to this version and give it some more testing. If we change it now, we'll get fewer eyes on the version from people who don't want to update their code.

Then there's nothing wrong with changing the method names in another major version bump for the next release.

Reply:
Ok. I'm trying to keep track of the breaking changes in the PR description above and will move them into a changelog soon. 

The UOM change may need to be reverted back to the previous 1 item list, there are several places this had to be patched in the HA code which makes everything cleaner overall and preps for ISYv5, but could be a PITA for some other integrations to update. Either way we need to document better what UOM returns for what version of ISY firmware.